opm/simulators/flow/GenericOutputBlackoilModule.cpp

```diff
@@ -451,10 +451,19 @@ assignToSolution(data::Solution& sol)
             return;
         }
 
-        sol.insert(std::get<std::string>(entry),
-                   std::get<UnitSystem::measure>(entry),
-                   std::move(std::get<2>(entry)),
-                   target);
+        if constexpr (std::is_same_v<Scalar,double>) {
+            sol.insert(std::get<std::string>(entry),
+                       std::get<UnitSystem::measure>(entry),
+                       std::move(std::get<2>(entry)),
+                       target);
+        } else {
+            const auto& input = std::get<2>(entry);
+            std::vector<double> data(input.begin(), input.end());
+            sol.insert(std::get<std::string>(entry),
+                       std::get<UnitSystem::measure>(entry),
+                       std::move(data),
+                       target);
```

Comment:
Minor nit: [`Solution::insert()`]() takes a `vector<double>` by value.  Could you not call this as
```C++
const auto& input = std::get<2>(entry);
solution.insert(..., std::vector<double>(input.begin(), input.end()), ...);
```
instead?

Reply:
sure.